md/bdm-access-control.md

```diff
@@ -314,3 +315,157 @@ Deploy your access control file.
+
+| Name | Type |
+|---|---|
+| fullname | STRING |
+
+2. an object Request with attributes:
+
+ | Name | Type |
+|---|---|
+| subject | STRING |
+| medicalComment | STRING |
+| content | STRING |
+| student | Student |
+
+Subject and Student are mandatory fields.
+We also define a custom query on the Request object, *findBySubject*:
+
+```
+SELECT r
+FROM Request r
+WHERE r.subject= :subject
+ORDER BY r.persistenceId
+
+```
+
+**2. Define the profiles**
+
+We will need a few new profiles, mapped to one or more users. We need one profile per subject, as well as a **Teacher** profile (for all the teachers). A maths teacher would therefore have **MathematicsTeacher** and a **Teacher** profile. In the rest of the guide we will use 3 profiles **MathematicsTeacher**, **PhysicsTeacher** and **Teacher**, mapped to **helen.kelly**, **jan.fischer** and both respectively.
+For the details on how to easily create and map those profiles see the [Profile Creation](profileCreation.md) page.
+
+**3. Create a process and generate some request instances**
+
+You will need some instances of the request object, as well as some students. To create them, follow the steps discribed in the section  [II - Invoice](#bdmFilling).
+
+**4. Create the visualisation application**
+
+We will now create an application for the teachers to review the student requests : 
+
+In the UI Designer, create an application page (*reviewRequests*):
+
+- Create 2 new variables
+
+    - **Name** : requestList
+        - **Type** : External API
+        - **API URL**: ../API/bdm/businessData/com.company.model.Request?q=findBySubject&p=0&c=10&f=subject%3D{{selectedSubject}}
+    
+    -  **Name** : selectedSubject
+	    -  **Type** : String
+    
+
+- Add a new Select box to the page (To choose beetween subjects):
+    - **Label** : Subject class
+    - **Available Values** : Mathematics, Physics (constants).  
+    - **Value** : selectedSubject    
+    
+    
+- Add  a new Table widget to the page (To display the requests):
+    - **Headers** : Id, Subject, Content, Medical comment, Student (constants)
+    - **Content** : requestList (script, click the fx icon to switch from contstant to script)
+    - **Column keys** : persistenceId, subject, content, medicalComment, student.fullname
+    
+Now in the Studio, create a new [application](applicationCreation.md). Call it *TeacherApp*, give it a theme, and a homepage token, and map it to the *Teacher* profile.
+Then create a one-page menu *Student Request*, with your homepage token and add to it the page you just created in the UID.
+
+If you preview the page at this point, both jan.fischer and helen.kelly can access your new application, select from a drop down list either *Mathematics* or *Physics*, wich should display all the Mathematics and all the Physics requests, respectively, for both users.
+
+**5. Security implementation**
+
+- Got to *BonitaStudioSubscription-7.7.0/workspace/tomcat/setup/*
+- Run *setup pull*
```

Comment:
* Something is missing, the user should set the path to the H2 directory before to run setup pull
* It would be nice to add the link to this page, just in case the user wonders what's this setup pull stuff and what does it do: https://documentation.bonitasoft.com/bonita/7.6/BonitaBPM_platform_setup
* There is a well detailed example on how to configure the dynamic security in the rest api authorization documentation, especially about all the stuff in the platform conf folder. it could be interesting to copy / link it here , to help the user to understand what's going on. https://documentation.bonitasoft.com/bonita/7.6/rest-api-authorization